spv/sync.go

```diff
@@ -54,9 +56,10 @@ type Syncer struct {
 	//
 	// TODO: Replace precise rescan filter with wallet db accesses to avoid
 	// needing to keep all relevant data in memory.
-	rescanFilter *wallet.RescanFilter
-	filterData   blockcf.Entries
-	filterMu     sync.Mutex
+	rescanFilter   map[int]*wallet.RescanFilter
+	filterData     map[int]*blockcf.Entries
```

Comment:
This wasn't a pointer to `blockcf.Entries`. Any reason for this change to pointer?